downstairs/src/region.rs

```diff
@@ -604,13 +612,23 @@ impl Extent {
 
         // XXX: schema updates?
 
-        Ok(Extent {
+        let extent = Extent {
             number,
             read_only,
             block_size: def.block_size(),
             extent_size: def.extent_size(),
-            inner: Some(Mutex::new(Inner { file, metadb })),
-        })
+            inner: Some(Mutex::new(Inner {
+                file,
+                metadb,
+                dirty_blocks: HashMap::new(),
+            })),
+        };
+
+        // Clean out any irrelevant block contexts, which may be present if downstairs
+        // crashed between a write() and a flush().
+        extent.fully_rehash_and_clean_all_stale_contexts(false)?;
```

Comment:
This will remove any contexts not associated with any data on disk. So yeah this could result in a change in the data. To cause this to happen you would need to

- write data to a read-write region
- crash or otherwise shut down downstairs before a flush succeeds
- mark it as read-only before opening the region again
- open the region